### PR TITLE
Split Refactoring.cpp into multiple files

### PR DESCRIPTION
**Pull request by ahoppen**
No code change, it’s just splitting the file.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

File: lib/Refactoring/ExtractExprBase.h [hunk 1]
```diff
@@ -47,6 +47,9 @@ class ExtractCheckResult {
   }
 };
 
+/// Check whether a given range can be extracted.
+/// Return true on successful condition checking,.
+/// Return false on failed conditions.
```
Comment (bnbarham) [inline lib/Refactoring/ExtractExprBase.h:55]: Was this extract from somewhere or new? It doesn't actually return a bool

Comment (ahoppen) [reply lib/Refactoring/ExtractExprBase.h:55]: Came from here

https://github.com/apple/swift/blob/df4f824b02947ca017f3c722d7803d222e52830f/lib/Refactoring/Refactoring.cpp#L1116-L1118

Comment (bnbarham) [reply lib/Refactoring/ExtractExprBase.h:55]: :(

**Review by bnbarham — APPROVED** [review]
Turns out we have one or two refactorings 😅

**Pull request merged**